Prototyping/Sit/sit.js

```diff
@@ -0,0 +1,692 @@
+            });
+        },
+        canSitDesktop: function () {
+            if (!chairProperties) this.setChairProperties();
+            var distanceFromSeat = Vec3.distance(MyAvatar.position, chairProperties.position);
+            return distanceFromSeat < SITTABLE_DISTANCE_MAX && !this.checkSeatForAvatar();
+        },
+
+        calculatePinHipPosition: function () {
+            if (!chairProperties) this.setChairProperties();
+
+            // get chairProperties to calculate middle of chair
+            seatCenterPosition = chairProperties.position;
+
+            var yOffset = chairProperties.dimensions.y / 2 * chairOffsetRatio;
+            seatCenterPosition = {
+                x: chairProperties.position.x,
+                y: chairProperties.position.y + yOffset,
+                z: chairProperties.position.z
+            };
+        },
+
+        setHeadToHipsDistance: function () { // used in update checks for Avatar Spine Error
+            // get hips world pos while sitting
+            if ((MyAvatar.getJointIndex("Head") === -1) || (MyAvatar.getJointIndex("Hips") === -1)) {
+                // this can probably be adjusted to be more accurate
+                headToHipsDistance = MyAvatar.getHeight() * HALF;
+            } else {
+                var headIndex = MyAvatar.getJointIndex("Head");
+                var hipsIndex = MyAvatar.getJointIndex("Hips");
+                var headTranslation = MyAvatar.getAbsoluteDefaultJointTranslationInObjectFrame(headIndex);
+                var hipsTranslation = MyAvatar.getAbsoluteDefaultJointTranslationInObjectFrame(hipsIndex);
+                headToHipsDistance = Math.abs(headTranslation.y - hipsTranslation.y);
+            }
+        },
+
+        setChairProperties: function () {
+            chairProperties = Entities.getEntityProperties(entityID, ["dimensions", "registrationPoint", "position", "rotation"]);
+        }
+
+    };
+
+    // Preload the animation file
+    this.animation = AnimationCache.prefetch(ANIMATION_URL);
+
+    this.preload = function (id) {
+        entityID = id;
+
+        overlays.preSit.preload();
+        utils.setChairProperties();
+    };
+
+    this.unload = function () {
+        if (Settings.getValue(SETTING_KEY) === entityID) {
+            standUp();
+        }
+
+        if (overlayIntervalTransparency !== null) {
+            Script.clearInterval(overlayIntervalTransparency);
+            overlayIntervalTransparency = null;
+        }
+
+        if (overlayCheckForHover !== null) {
+            Script.clearInterval(overlayCheckForHover);
+            overlayCheckForHover = null;
+        }
+
+        if (overlaySittable) overlays.sittable.remove();
+        if (overlayPreSit) overlays.preSit.remove();
+        if (overlayStandUp) overlays.standUp.remove();
+
+    };
+
+    function sitAndPinAvatar() {
+
+        Settings.setValue(SETTING_KEY, entityID);
+        sitDownSettlePeriod = Date.now() + IK_SETTLE_TIME;
+
+        print(4);
```

Comment:
please remove these debug statements.  Or add a global variable SITTING_DEBUG and do something like:
```
if (SITTING_DEBUG) {
    print("got here 4");
}
```